src/metrics_rs.rs

```diff
@@ -0,0 +1,221 @@
+                )*
+            }
+        }
+
+        #[test]
+        fn test_no_fields_missing() {
+            // test that no fields are missing. We can't use exhaustive matching here
+            // since the metrics structs are #[non_exhaustive], so use a debug impl
+            let debug = format!("{:#?}", <$metrics_name>::default());
+            for line in debug.lines() {
+                if line == format!("{} {{", stringify!($metrics_name)) || line == "}" {
+                    continue
+                }
+                $(
+                    let expected = format!("    {}:", stringify!($ignore));
+                    if line.contains(&expected) {
+                        continue
+                    }
+                );*
+                $(
+                    let expected = format!("    {}:", stringify!($name));
+                    eprintln!("{}", expected);
+                    if line.contains(&expected) {
+                        continue
+                    }
+                );*
+                $(
+                    let expected = format!("    {}:", stringify!($unstable_name));
+                    eprintln!("{}", expected);
+                    if line.contains(&expected) {
+                        continue
+                    }
+                );*
+                panic!("missing metric {:?}", line);
+            }
+        }
+    }
+}
+
+pub(crate) use capture_metric_ref;
+pub(crate) use describe_metric_ref;
+pub(crate) use kind_to_type;
+pub(crate) use metric_key;
+pub(crate) use metric_refs;
+
+pub(crate) trait MyMetricOp<T> {
+    fn op(self, t: T);
+}
+
+impl<T> MyMetricOp<T> for (&metrics::Counter, Duration) {
+    fn op(self, _: T) {
+        self.0
+            .increment(self.1.as_micros().try_into().unwrap_or(u64::MAX));
+    }
+}
+
+impl<T> MyMetricOp<T> for (&metrics::Counter, u64) {
+    fn op(self, _t: T) {
+        self.0.increment(self.1);
+    }
+}
+
+impl<T> MyMetricOp<T> for (&metrics::Gauge, Duration) {
+    fn op(self, _t: T) {
+        self.0.set(self.1.as_micros() as f64);
+    }
+}
+
+impl<T> MyMetricOp<T> for (&metrics::Gauge, u64) {
+    fn op(self, _: T) {
+        self.0.set(self.1 as f64);
+    }
+}
+
+impl<T> MyMetricOp<T> for (&metrics::Gauge, usize) {
+    fn op(self, _t: T) {
+        self.0.set(self.1 as f64);
+    }
+}
+
+#[cfg(tokio_unstable)]
+impl MyMetricOp<&tokio::runtime::RuntimeMetrics> for (&metrics::Histogram, Vec<u64>) {
+    fn op(self, tokio: &tokio::runtime::RuntimeMetrics) {
+        for (i, bucket) in self.1.iter().enumerate() {
+            let range = tokio.poll_time_histogram_bucket_range(i);
+            if *bucket > 0 {
+                // emit using range.start to avoid very large numbers for open bucket
+                // FIXME: do we want to do something else here?
+                self.0
+                    .record_many(range.start.as_micros() as f64, *bucket as usize);
+            }
+        }
+    }
+}
```

Comment:
To be honest, I was a little confused by this `impl`. It seems like it's specific to a single histogram metric in the runtime. Does this make it impossible to add any other histogram metrics in the future?